providers/azure/resources/iam.go

```diff
@@ -164,3 +159,180 @@ func azureToMqlPermission(runtime *plugin.Runtime, id string, permission *author
 	}
 	return p, nil
 }
+
+func (a *mqlAzureSubscriptionAuthorizationService) roleAssignments() ([]interface{}, error) {
+	conn := a.MqlRuntime.Connection.(*connection.AzureConnection)
+	token := conn.Token()
+	subId := a.SubscriptionId.Data
+
+	var client, err = authorization.NewRoleAssignmentsClient(subId, token, &arm.ClientOptions{
+		ClientOptions: conn.ClientOptions(),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := context.Background()
+	// we're interested in subscription-level role definitions, so we scope this to the subscription,
+	// on which this connection is running
+	pager := client.NewListForSubscriptionPager(&authorization.RoleAssignmentsClientListForSubscriptionOptions{})
+	res := []interface{}{}
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return nil, err
+		}
+		for _, roleAssignment := range page.Value {
+			mqlRoleAssignment, err := newMqlRoleAssignment(a.MqlRuntime, roleAssignment)
+			if err != nil {
+				return nil, err
+			}
+			res = append(res, mqlRoleAssignment)
+		}
+	}
+	return res, nil
+}
+
+type mqlAzureSubscriptionAuthorizationServiceRoleAssignmentInternal struct {
+	roleDefinitionId string
+}
+
+func newMqlRoleAssignment(runtime *plugin.Runtime, roleAssignment *authorization.RoleAssignment) (*mqlAzureSubscriptionAuthorizationServiceRoleAssignment, error) {
+	r, err := CreateResource(runtime, "azure.subscription.authorizationService.roleAssignment",
+		map[string]*llx.RawData{
+			"__id":        llx.StringDataPtr(roleAssignment.ID),
+			"id":          llx.StringDataPtr(roleAssignment.Name), // name is the id :-)
```

Comment:
why not have this named `name`?

Reply:
Because the name is not the id :-) Do not ask me why.